### ISSUE TITLE
Error accessing GCP artifacts when using special characters in task name

### ISSUE THREAD

**Issue by materight** [allegroai/clearml#1051]
## Describe the bug
When uploading to GCP and using special characters in the task name, e.g. `:`, the artifact can't be retrived correctly. The returned artifact url is quoted, but GCP requires the url to be unquoted. Adding a call to `urllib.parse.unquote(artifact._url)` fixes the issue.

## To reproduce
```python
import urllib
from clearml import Task

task = Task.init(project_name='TEST', task_name='test:1')
task.upload_artifact('test_artifact', {'a': 1}, wait_on_upload=True)
artifact = task.artifacts['test_artifact']
#artifact._url = urllib.parse.unquote(artifact._url) # Fix
val = artifact.get(force_download=True)
print(val)
```

## Expected behaviour
The artifact should be downloaded correctly. Without the fix, it tries to access:
`gs://cerrion-clearml/TEST/test%253A1.7c3dc2f40455456e88cf4c154edb965c/artifacts/test_artifact/test_artifact.json`
while the correct url would be without the `%25`:
`gs://cerrion-clearml/TEST/test%3A1.7c3dc2f40455456e88cf4c154edb965c/artifacts/test_artifact/test_artifact.json`

## Environment
* Server type: self hosted
* ClearML SDK Version: 1.11.0
* ClearML Server Version: WebApp: 1.11.0-373 • Server: 1.11.0-373 • API: 2.25
* Python Version: 3.10.10
* OS: Linux

**Comment by ainoam** [comment]
Thanks for reporting @materight.

Hope to have a fix out soon.

**Comment by materight** [comment]
Great! Thanks for the update

**Comment by materight** [comment]
Fixed in v1.12.0
